File: LeetCode/C/94-tree-二叉树的中序遍历.cpp
/*
 * @lc app=leetcode.cn id=94 lang=cpp
 *
 * [94] 二叉树的中序遍历
 */
#include "main.h"
// @lc code=start
class Solution {
  public:
    vector<int> inorderTraversal(TreeNode *root) {
        vector<int> res;

        function<void(TreeNode *)> in = [&](TreeNode *node) {
            if (node == nullptr)
                return;
            in(node->left);
            res.push_back(node->val);
            in(node->right);
        };

        in(root);
        return res;
    }
};
// @lc code=end

File: LeetCode/Go/94-tree-二叉树的中序遍历.go
/*
 * @lc app=leetcode.cn id=94 lang=golang
 *
 * [94] 二叉树的中序遍历
 */
// package leetcode

// @lc code=start
func inorderTraversal(root *TreeNode) []int {
	res := []int{}

	var inorder func(*TreeNode)
	inorder = func(root *TreeNode) {
		if root == nil {
			return
		}
		inorder(root.Left)
		res = append(res, root.Val)
		inorder(root.Right)
	}

	inorder(root)
	return res
}

// @lc code=end
